.github/workflows/_meta_mac_portable.yaml

```
@@ -0,0 +1,117 @@
+on:
+  workflow_call:
+    inputs:
+      release:
+        required: false
+        default: false
+        type: boolean
+    secrets:
+      deploy-host:
+        required: false
+      deploy-user:
+        required: false
+      deploy-key:
+        required: false
+
+jobs:
+  build:
+    name: 'Build Portable FFmpeg'
+    runs-on: ${{ matrix.os.name }}
+    strategy:
+      fail-fast: true
+      matrix:
+        # Currently, macOS 12 is x86 exclusive and macOS 14 is arm exclusive and we have no other way to specify arch
+        os:
+          - name: macos-12
+            arch: x86_64
+          - name: macos-14
+            arch: arm64
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Prepare prefix dir
+        run: |
+          sudo mkdir /opt/ffbuild
+          sudo chmod 777 /opt/ffbuild
+
+      - name: Build Portable
+        run: ./builder/buildmac.sh
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        with:
+          name: mac-${{ matrix.os.arch }}-portable
+          path: ./builder/artifacts
+
+  publish:
+    name: Publish Portable Release
+    if: ${{ inputs.release }}
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    strategy:
+      fail-fast: true
+      matrix:
+        arch: [arm64, x86_64]
+
+    steps:
+      - name: Set Versions
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        id: set_version
+        with:
+          script: |
+            const tag = context.ref.substring(10)
+            const no_v = tag.replace('v', '')
+            const dash_index = no_v.lastIndexOf('-')
+            const no_dash = (dash_index > -1) ?  no_v.substring(0, dash_index) : no_v
+            core.setOutput('tag', tag)
+            core.setOutput('no-v', no_v)
+            core.setOutput('no-dash', no_dash)
+
+      - name: Download Artifacts
+        uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427 # v4.1.4
+        with:
+          name: mac-${{ matrix.arch }}-portable
+          path: artifact
+
+      - name: Prepare Release Assets
+        run: |-
+          pushd artifact
+          find * -type f \( -name "*.xz" -o -name "*.zip" \) | while read file; do
```

```suggestion
          find * -type f \( -name "*.xz" -o -name "*.zip" \) | while read file; do
```

Isn't the builder producing tar archives?

---

This is copied from our other release pipeline, and the builder produces tar.xz archives. I've seen our other build scripts producing similar checksum files as well, does that mean this step is redundant?

---

If we already have sha256sum files it is redundant. But I think we do not create them for windows.